azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageVersionDaoImpl.java

```diff
@@ -128,30 +157,103 @@ public int createImageVersion(ImageVersion imageVersion) {
     return imageVersions;
   }
 
+  @Override
+  public Optional<ImageVersion> getImageVersion(String imageTypeName, String imageVersion,
+      State versionState) throws ImageMgmtException {
+    ImageMetadataRequest imageMetadataRequest = ImageMetadataRequest.newBuilder()
+        .addParam(IMAGE_TYPE, imageTypeName)
+        .addParam(IMAGE_VERSION, imageVersion)
+        .addParam(VERSION_STATE, versionState)
+        .build();
+    List<ImageVersion> imageVersions = findImageVersions(imageMetadataRequest);
+    return imageVersions != null && imageVersions.size() > 0 ? Optional.of(imageVersions.get(0))
+        : Optional.empty();
+  }
+
+  @Override
+  public List<ImageVersion> getActiveVersionByImageTypes(Set<String> imageTypes)
+      throws ImageMgmtException {
+    List<ImageVersion> imageVersions = new ArrayList<>();
+    try {
+      // Add outer select clause
+      StringBuilder inClauseBuilder = new StringBuilder();
+      for (int i = 0; i < imageTypes.size(); i++) {
+        inClauseBuilder.append("?,");
+      }
+      inClauseBuilder.setLength(inClauseBuilder.length() - 1);
+      Map<String, String> valueMap = new HashMap<>();
+      valueMap.put("image_types", inClauseBuilder.toString());
+      StrSubstitutor strSubstitutor = new StrSubstitutor(valueMap);
+      String query = strSubstitutor.replace(SELECT_LATEST_ACTIVE_IMAGE_VERSION_QUERY);
+      log.info("Image version getActiveVersionByImageTypes query : " + query);
+      List<Object> params = new ArrayList<>();
+      params.add(State.ACTIVE.getStateValue());
+      params.addAll(imageTypes);
+      params.add(State.ACTIVE.getStateValue());
```

Comment:
Why there is duplicate lines?

Reply:
There are two inner queries in both the State column is referred. Hence, two same params needs to be substituted.